source/app/api.py

```diff
@@ -258,38 +266,35 @@ def asd_delete(disk_id, asd_id):
         raise BadRequest('Disk {0} is not yet initialized'.format(alias))
 
     @staticmethod
-    @get('/update/information')
-    def get_update_information():
+    @get('/update/package_information')
+    def get_package_information():
         """ Retrieve update information """
         with file_mutex('package_update'):
             API._logger.info('Locking in place for package update')
-            return UpdateController.get_update_information()
+            return SDMUpdateController.get_package_information()
 
     @staticmethod
-    @post('/update/execute/<status>')
-    def execute_update(status):
+    @post('/update/execute')
+    def update():
         """
         Execute an update
-        :param status: Current status of the update
-        :type status: str
         """
         with file_mutex('package_update'):
-            return UpdateController.execute_update(status)
+            return SDMUpdateController.update()
```

Comment:
Can you track progress in any way? If this call takes too long the client will timeout, so you need some async system.